lib/presentation/ui/pages/sign_up_page.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_target/constants.dart';
+import 'package:flutter_target/data/services/user_service.dart';
+import 'package:flutter_target/domain/cubit/sign_up_cubit.dart';
+import 'package:flutter_target/presentation/ui/pages/sign_in_page.dart';
+import 'package:flutter_target/presentation/ui/widgets/text_form_custom.dart';
+
+class SignUpPage extends StatelessWidget {
+  static const String id = 'sign_up';
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => SignUpCubit(UserService()),
+      child: Scaffold(
+        backgroundColor: Colors.white,
+        body: SingleChildScrollView(
+          physics: ClampingScrollPhysics(),
+          child: Form(
+            child: Stack(
+              alignment: Alignment.topCenter,
+              children: [
+                Image.asset(
+                  kCircleLoginImage,
+                ),
+                Padding(
+                  padding: EdgeInsets.symmetric(vertical: 100.0),
+                  child: Text(
+                    kAppName,
+                    style: Theme.of(context).textTheme.titleLarge,
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+                Container(
+                  margin: EdgeInsets.only(top: 180),
+                  padding: EdgeInsets.symmetric(horizontal: 50.0),
+                  child: BlocBuilder<SignUpCubit, SignUpState>(
+                    builder: (context, state) {
+                      return Column(
+                        children: [
+                          TextFormCustom(
+                            labelText: kName,
+                            onTextChanged: (value) {
+                              context
+                                  .read<SignUpCubit>()
+                                  .usernameChanged(value);
+                            },
+                          ),
+                          TextFormCustom(
+                            labelText: kEmail,
+                            onTextChanged: (value) {
+                              context.read<SignUpCubit>().emailChanged(value);
+                            },
+                          ),
```

Comment:
Try this.
```suggestion
                          TextFormCustom(
                            labelText: kEmail,
                            onTextChanged: context.read<SignUpCubit>().emailChanged,
                          ),
```
same in the others

Reply:
Im gonna change it, ty!

Reply:
Im changing this, and I wonder how Dart know that the param should be pass to the method. Do you know?

Reply:
you are passing a `Function(String value)` that is what `onTextChanged` is expecting